presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -260,14 +256,18 @@ PlanNodeStatsEstimate calculateJoinComplementStats(
             return UNKNOWN_STATS;
         }
 
+        // TODO: add support for non-equality conditions (e.g: <=, !=, >)
+        int numberOfFilterClauses = filter.map(
+                exression -> ExpressionUtils.extractConjuncts(exression).size())
+                .orElse(0);
+
         // Heuristics: select the most selective criteria for join complement clause.
         // Principals behind this heuristics is the same as in computeInnerJoinStats:
         // select "driving join clause" that reduces matched rows the most.
         return IntStream.range(0, criteria.size())
                 .mapToObj(drivingClauseId -> {
                     EquiJoinClause drivingClause = criteria.get(drivingClauseId);
-                    List<EquiJoinClause> remainingClauses = copyWithout(criteria, drivingClauseId);
-                    return calculateJoinComplementStats(leftStats, rightStats, drivingClause, remainingClauses);
+                    return calculateJoinComplementStats(leftStats, rightStats, drivingClause, criteria.size() - 1 + numberOfFilterClauses);
```

Comment:
maybe instead of `-1`, use `remainingCluases.size()`?

Reply:
I didn't want to pass `remainingCluases` if I'm just using clauses list for getting size